src/Numerical/Spectral/DefiniteIntegral.hpp

```diff
@@ -0,0 +1,27 @@
+
+/// \file
+/// Defines function definite_integral.
+
+#pragma once
+
+#include <cstddef>
+
+class DataVector;
+template <size_t>
+class Index;
+
+/*! \ingroup Functors
+ * \brief Compute the definite integral of a grid-function over a manifold.
+ *
+ * The integral is computed on the reference element by multiplyting the
+ * DataVector
+ * with the LegendreGaussLobatto integration weights in that dimension.
+ *
+ * \requires number of points in f and extents are equal.
+ *
+ * \param f the grid function to integrate.
+ * \param extents the extents of the manifold on which f is located.
+ * \returns the definite integral of f on the manifold.
+ */
+template <size_t Dim>
+double definite_integral(const DataVector& f, const Index<Dim>& extents);
```

Comment:
I think we want this function in the `lgl` namespace as it evaluates the definite integral on the reference cell.  We are likely going to want another function that computes definite integrals in a frame other than the logical frame